server/src/main/java/org/opensearch/cluster/routing/OperationRouting.java

```diff
@@ -343,7 +344,14 @@ private ShardIterator preferenceActiveShardIterator(
         // for a different element in the list by also incorporating the
         // shard ID into the hash of the user-supplied preference key.
         routingHash = 31 * routingHash + indexShard.shardId.hashCode();
-        if (ignoreAwarenessAttributes()) {
+        if (weightedRoutingMetadata != null && weightedRoutingMetadata.getWeightedRouting().isSet() && isStrictWeightedShardRouting) {
+            return indexShard.activeInitializingShardsSimpleWeightedIt(
+                weightedRoutingMetadata.getWeightedRouting(),
+                nodes,
+                getWeightedRoutingDefaultWeight(),
+                routingHash
+            );
```

Comment:
how is this different from `preferAttributesActiveInitializingShardsIt` ?  If we are shuffling and not considering weights for now, we can as well use `preferAttributesActiveInitializingShardsIt`  .

Reply:
we are considering weights here with `preferAttributesActiveInitializingShardsIt`